### PR TITLE
Skeleton PKCS #11 demo project

### PR DESCRIPTION
**Pull request by lundinc2**

### CODE REVIEW

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/WIN32.vcxproj [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+  </PropertyGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
+    <Midl>
+      <TypeLibraryName>.\Debug/WIN32.tlb</TypeLibraryName>
+      <HeaderFileName>
+      </HeaderFileName>
+    </Midl>
+    <ClCompile>
+      <Optimization>Disabled</Optimization>
+      <AdditionalIncludeDirectories>\\Mac\Home\Documents\FreeRTOS-Labs\FreeRTOS-Plus\Source\FreeRTOS-Plus-PKCS11\include;..\..\..\FreeRTOS-Plus\Source\FreeRTOS-Plus-PKCS11\3rdparty\pkcs11;..\..\..\FreeRTOS-Plus\Source\FreeRTOS-Plus-PKCS11\3rdparty\mbedtls\include;..\..\..\FreeRTOS-Plus\Source\FreeRTOS-Plus-PKCS11\3rdparty\mbedtls_utils;..\..\..\FreeRTOS\Source\include;..\..\..\FreeRTOS\Source\portable\MSVC-MingW;.;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (lundinc2) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/WIN32.vcxproj]: Make this a relative path.

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/WIN32.vcxproj [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+      <AdditionalLibraryDirectories>..\Common\ethernet\lwip-1.4.0\ports\win32\WinPCap</AdditionalLibraryDirectories>
+      <AdditionalDependencies>wpcap.lib;%(AdditionalDependencies)</AdditionalDependencies>
+    </Link>
+    <Bscmake>
+      <SuppressStartupBanner>true</SuppressStartupBanner>
+      <OutputFile>.\Release/WIN32.bsc</OutputFile>
+    </Bscmake>
+  </ItemDefinitionGroup>
+  <ItemGroup>
+    <ClCompile Include="..\..\..\..\amazon-freertos\libraries\3rdparty\mbedtls\library\aes.c" />
```
Comment (lundinc2) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/WIN32.vcxproj]: Make this use the correct file path.

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/WIN32.vcxproj.filters [hunk 3]
```diff
@@ -0,0 +1,629 @@
...
+    <ClCompile Include="..\..\..\FreeRTOS\Source\timers.c">
+      <Filter>FreeRTOS\Source</Filter>
+    </ClCompile>
+    <ClCompile Include="..\..\..\FreeRTOS\Source\portable\MemMang\heap_4.c">
+      <Filter>FreeRTOS\Source\Portable</Filter>
+    </ClCompile>
+    <ClCompile Include="..\..\..\FreeRTOS\Source\portable\MSVC-MingW\port.c">
+      <Filter>FreeRTOS\Source\Portable</Filter>
+    </ClCompile>
+    <ClCompile Include="..\..\..\..\amazon-freertos\libraries\3rdparty\mbedtls\library\aes.c">
```
Comment (lundinc2) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/WIN32.vcxproj.filters]: Should not point to amazon-freertos.

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/aws_mbedtls_config.h [hunk 4]
```diff
@@ -0,0 +1,3324 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of mbed TLS (https://tls.mbed.org)
+ */
+
```
Comment (lundinc2) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/aws_mbedtls_config.h]: TODO: Consider removing the functionality not used in the PKCS #11 demos.

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/main.c [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+ * 1 tab == 4 spaces!
+ */
+
+/***
+ * See https://www.FreeRTOS.org/mqtt/index.html for configuration and usage instructions.
+ ***/
+
+/* Standard includes. */
+#include <stdio.h>
+#include <time.h>
```
Comment (lundinc2) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/main.c]: remove this

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/main.c [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+    BaseType_t xReturned;
+    TaskHandle_t xHandle = NULL;
+	xReturned = xTaskCreate( 
+            vPKCS11ManagementDemo, 
+            "PKCS11 Demo", 
+            configPKCS11_DEMO_STACK_SIZE, 
+            NULL, 
+            tskIDLE_PRIORITY + 1,
+            &xHandle );
+    configASSERT( xReturned == pdPASS );
```
Comment (lundinc2) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/main.c:68]: TODO: Uncrustify.

File: FreeRTOS-Plus/Source/FreeRTOS-Plus-PKCS11/3rdparty/mbedtls_config/README.md [hunk 7]
```diff
@@ -0,0 +1,2 @@
+## Background
```
Comment (lundinc2) [inline FreeRTOS-Plus/Source/FreeRTOS-Plus-PKCS11/3rdparty/mbedtls_config/README.md]: TODO: Consider deleting this folder.

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/FreeRTOSConfig.h [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+/* Assert call defined for debug builds. */
+#ifdef _DEBUG
+	extern void vAssertCalled( const char *pcFile, uint32_t ulLine );
+	#define configASSERT( x ) if( ( x ) == 0 ) vAssertCalled( __FILE__, __LINE__ )
+#endif /* _DEBUG */
+
+
+
+/* Application specific definitions follow. **********************************/
```
Comment (yuhui-zheng) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/FreeRTOSConfig.h:126]: Do you need these configs (line 126 -- line 198) for PKCS11 demo?

**Review by yuhui-zheng — APPROVED** [review]

**Pull request merged**